### PR TITLE
Add GPS status info with timeout setting / GitPod fix

### PR DESCRIPTION
**Pull request by szolnokit**
Add new option `--gps-timeout <milliseconds>` for detect "No GPS signal", default: 1 min
Added to GUI also.

Add some "information" variables for GPS status:
  `%GPSSTATUS%`: "LOST" if  GPS timeout happened, otherwise "OK"
  `%GPSDATETIME%`: Last valid GPS position time (empty if no last time)
  `%GPSDIFFTIME%`: Time elapsed while last GPS position
  `%GPSLOSTTIME%`: Time elapsed while GPS lost (empty if GPS OK)

Also, added the missing `--information <text>` CLI option.

Demo: https://youtu.be/6TwCpLMo6-w

### CODE REVIEW

**Comment by szolnokit** [comment]
Also, added the missing --information <text> CLI options.

**Comment by szolnokit** [comment]
GitPod working now out of the box.
Bulids and tests are 100%
Container updated to Java 17 JDK
Debbuger working
No yellow GUI problem
Screenshot_2

**Comment by McPringle** [comment]
Sorry for the review delay, I was extremely busy (new release at work and Diploma exams of my students). I'll review your PR today!

File: .gitpod.Dockerfile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 FROM gitpod/workspace-full-vnc
 USER root
 RUN apt-get update \
-    && bash -c ". /home/gitpod/.sdkman/bin/sdkman-init.sh && sdk install java 16.0.1.hs-adpt" \
+    && bash -c ". /home/gitpod/.sdkman/bin/sdkman-init.sh && sdk install java 17.0.1-zulu" \
+    && bash -c 'echo export JAVA_TOOL_OPTIONS=\"\$JAVA_TOOL_OPTIONS -Dsun.java2d.xrender=false\" >> /home/gitpod/.bashrc' \
```
Comment (McPringle) [inline .gitpod.Dockerfile:5]: Nested `JAVA_TOOL_OPTIONS`? Let's try it... :-)

File: src/main/java/app/gpx_animator/core/Help.java [hunk 2]
```diff
@@ -31,7 +31,7 @@ private Help() throws InstantiationException {
         throw new InstantiationException("Help is a Utility class and can't be instantiated!");
     }
 
-    @SuppressWarnings("DuplicateStringLiteralInspection")
+    @SuppressWarnings("PMD.AvoidDuplicateLiterals")
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/Help.java:34]: I think both suppressions are needed:
```java
@SuppressWarnings({ "DuplicateStringLiteralInspection", "PMD.AvoidDuplicateLiterals" })
```

File: src/main/java/app/gpx_animator/core/Help.java [hunk 3]
```diff
@@ -94,8 +95,9 @@ public static void printHelp(final OptionHelpWriter w) {
         w.writeOptionHelp(Option.VIEWPORT_HEIGHT, "viewport-height", false, cfg.getViewportHeight()); //NON-NLS
         w.writeOptionHelp(Option.VIEWPORT_INERTIA, "viewport-inertia", false, cfg.getViewportInertia()); //NON-NLS
         w.writeOptionHelp(Option.WAYPOINT_SIZE, "size", false, cfg.getWaypointSize()); //NON-NLS
-        w.writeOptionHelp(Option.WIDTH, "width", false, "(800)"); // TODO cfg.getWidth() NON-NLS
+        w.writeOptionHelp(Option.WIDTH, "width", false, cfg.getWidth()); //NON-NLS
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/Help.java:98]: If I remember correctly, there was a problem using `getWidth()` based on if the width was set by the user or calculated programmatically. We should test that very well.

File: src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java [hunk 4]
```diff
@@ -85,12 +88,30 @@ public void renderFrame(final int frame, @Nullable final Point2D marker, @NonNul
         final var latLongString = getLatLonString(marker);
         final var speedString = getSpeedString(marker, time, frame);
 
-        final var text = information
-                .replace("%SPEED%", speedString)
-                .replace("%LATLON%", latLongString)
-                .replace("%DATETIME%", dateTimeString);
-
+        final var gpsTime = getTime(marker);    //TODO --Get strings from resource
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java:91]: This `TODO` should be resolved, it's an easy one.

File: src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java [hunk 5]
```diff
@@ -85,12 +88,30 @@ public void renderFrame(final int frame, @Nullable final Point2D marker, @NonNul
...
+        var gpsDateTimeString = "Unknown";
+        var gpsDiffTime = 0L;
+        var gpsDiffTimeString = "";
+        var gpsStatus = false;
+        var gpsLostTimeString = "";
+        var gpsStatusString = "NOTIME";           //TODO --Get strings from resource
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java:97]: This `TODO` should be resolved, it's an easy one.

File: src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java [hunk 6]
```diff
@@ -85,12 +88,30 @@ public void renderFrame(final int frame, @Nullable final Point2D marker, @NonNul
...
+        if (gpsTime > 0) {
+            gpsDiffTime = time - gpsTime;
+            gpsDateTimeString = dateFormat.format(gpsTime);
+            gpsDiffTimeString = DurationFormatUtils.formatDuration(gpsDiffTime, "+H:mm:ss");
+            gpsStatus = gpsTimeout <= 0 || gpsDiffTime < gpsTimeout;
+            gpsLostTimeString = gpsStatus ? "" : gpsDiffTimeString;
+            gpsStatusString = gpsStatus ? "OK" : "LOST";      //TODO --Get strings from resource
```
Comment (McPringle) [inline src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java:104]: This `TODO` should be resolved, it's an easy one.

**Review by McPringle — APPROVED** [review]
Thank you very much for your contribution!

File: src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java (hunk 6 above)
Comment (szolnokit) [reply src/main/java/app/gpx_animator/core/renderer/plugins/InformationPlugin.java:104]: Not easy, bacuse the plugin don't know the language resoure object.

File: src/test/java/app/gpx_animator/ui/cli/CommandLineIT.java [hunk 7]
```diff
@@ -61,8 +61,8 @@ void testBasicCommandLine() throws Exception {
         assertDone();
 
         final var fileSize = new File(outputFile).length();
-        assertTrue(fileSize > 250_000, "Output file size (%s bytes) too small, check content".formatted(fileSize));
-        assertTrue(fileSize < 350_000, "Output file size (%s bytes) too big, check content".formatted(fileSize));
+        assertTrue(fileSize > 300_000, "Output file size (%s bytes) too small, check content".formatted(fileSize));
+        assertTrue(fileSize < 400_000, "Output file size (%s bytes) too big, check content".formatted(fileSize));
```
[the remark this answers is not in the document]
Comment (szolnokit) [reply src/test/java/app/gpx_animator/ui/cli/CommandLineIT.java:65]: I think, the sze will be increased because the addtional texts on the video

**Review by McPringle — APPROVED** [review]

**Comment by McPringle** [comment]
Sorry for merging this PR so late - I was under high stress the last few months and totally forgot about this one!

**Pull request merged**